contracts/tansu/src/contract_tansu.rs

```diff
@@ -150,6 +161,8 @@ impl TansuTrait for Tansu {
         new_admins_config: Option<types::AdminsConfig>,
     ) {
         auth_admin(&env, &admin);
+        // Proposing an upgrade mutates state; require not paused.
+        Self::require_not_paused(env.clone());
```

Comment:
Even more important here. Admins would even pause the contract to do a migration to prevent anything bad from happening.